### PR TITLE
Use hdt-cpp new `skip` function

### PR DESCRIPTION
**Pull request by dinikolop**
This uses the new hdt-cpp `skip` function, instead of the (now absolute) `goTo` function. Guess you can cherrypick everything here except the gitsubmodule reference (which points to the opentriply fixGoto branch for testing purposes until https://github.com/rdfhdt/hdt-cpp/pull/89 is merged and released to master).
Also added some more tests, particularly to verify whether the ObjectIndexIterator (for `? ? <iri>` queries) still works as expected

### CODE REVIEW

File: test/hdt-test.js [hunk 1]
```diff
@@ -231,6 +231,7 @@ describe('hdt', function () {
         });
       });
 
+
```
Comment (RubenVerborgh) [inline test/hdt-test.js]: Probably don't need this one.

File: test/hdt-test.js [hunk 2]
```diff
@@ -260,6 +261,60 @@ describe('hdt', function () {
...
+      describe('with pattern ex:s2 ex:p1 null', function () {
+        var triples;
+        before(function (done) {
+          document.searchTriples('http://example.org/s2', 'http://example.org/p1', null,
+                          function (error, t, c, e) { triples = t; done(error); });
+        });
+
+        it('should return an array with matches', function () {
+          triples.should.be.an.Array;
+          triples.should.have.lengthOf(10);
```
Comment (RubenVerborgh) [inline test/hdt-test.js]: Nitpick: I'm using `length()` throughout.

File: test/hdt-test.js [hunk 3]
```diff
@@ -260,6 +261,60 @@ describe('hdt', function () {
...
+          triples.should.be.an.Array;
+          triples.should.have.lengthOf(1);
+          triples[0].should.eql({
+            subject:   'http://example.org/s2',
+            predicate: 'http://example.org/p1',
+            object:    'http://example.org/o010' });
+        });
+      });
+      // Use this pattern to check whether the ObjectIndexIterator implementation
+      // in hdt-cpp works
```
Comment (RubenVerborgh) [inline test/hdt-test.js]: Maybe add a reference to the issue URL? I prefer having an empty line in between `describe` blocks as well.

**Comment by RubenVerborgh** [comment]
Thanks so much, saves me a lot of work! I'll wait for the commit on hdt-cpp to land in master, so we can release a new hdt-cpp.

**Comment by RubenVerborgh** [comment]
Thanks, will check soon.

File: test/hdt-test.js (hunk 2 above)
Comment (dinikolop) [reply test/hdt-test.js]: @RubenVerborgh I found only one use of length() in the hdt-test.js and lengthOf() is used in all other cases. I changed all of them into length(). Is that what you meant? Thank you! :)

**Pull request merged**